modules/core/include/opencv2/core/types.hpp

```diff
@@ -63,6 +63,64 @@ namespace cv
 //! @addtogroup core_basic
 //! @{
 
+//////////////////////////////// Enum operators //////////////////////////////
+
+template<typename T, typename T2, typename std::enable_if< std::is_enum<T>::value
+    && T::CV_FLAGS_ENUM == T::CV_FLAGS_ENUM >::type* = nullptr>
+static T operator|(const T2& a, const T& b)
+{
```

Comment:
How about utilizing SFINAE and introducing `CV_FLAGS_ENUM` in the flag-based enums? I guess this solution is elegant enough.